src/plugins/Miner.cpp

```diff
@@ -138,7 +138,17 @@ void DistrubuteMineralWorker(const sc2::Unit* unit_) {
 
 }  // namespace
 
+Miner::Miner() {
+    m_turns_to_skip = 7;
+}
+
 void Miner::OnStep(Builder* builder_) {
+    if (m_skip_until_turn > gAPI->observer().GetGameLoop())
```

Comment:
My idea was to move this check out on the Dispatchers level. Dispatcher calls the plugins so it can check when it is time to wake this or that plugin. Also, it allows to make flow control centralised and write less code.
What do you think about moving this code there?

Reply:
I kind of wanted the Dispatcher to not need to know the internals of how plugins work, but yes it does centralize the flow control. But then I will need to add a Getter to a new Plugin.cpp, as well as the default constructor to make sure the skip variables aren't forgotten and left uninitialized, unless that's acceptable and the author's responsibility to write the inherited constructors correctly. I was mostly just trying to avoid making a Plugin.cpp and have to deal with adding it to all the project/makefile stuff.

Reply:
Dispatcher doesn't know the internals, but it handle the application lifecycle. Consider it being a nerve system which triggers different body organs via signals. It doesn't know how anything works but it spreads the reaction of different events and can manipulate the frequency of plugins callbacks, not the internal logic.

> I was mostly just trying to avoid making a Plugin.cpp and have to deal with adding it to all the project/makefile stuff.

I don't think there will be any problem, if you add Plugin.cpp. This is fairly common approach.